dnf5/commands/download/download.cpp

```diff
@@ -179,6 +194,36 @@ void DownloadCommand::run() {
         return;
     }
 
+    if (srpm_option->get_value()) {
+        std::map<std::string, libdnf5::rpm::Package> source_pkgs;
+
+        libdnf5::rpm::PackageQuery source_pkg_query(ctx.base);
+        source_pkg_query.filter_arch({"src"});
+        source_pkg_query.filter_available();
+
+        for (auto & [nevra, pkg] : download_pkgs) {
+            auto sourcerpm = pkg.get_sourcerpm();
+
+            if (!sourcerpm.empty()) {
+                libdnf5::rpm::PackageQuery pkg_query(source_pkg_query);
+                // Remove ".rpm" to get sourcerpm nevra
+                sourcerpm.erase(sourcerpm.length() - 4);
+                pkg_query.resolve_pkg_spec(sourcerpm, {}, true);
+
```

Comment:
One last thing, I have noticed the `sourcerpm` doesn't include epoch. Could you add a filter for it?
```cpp
pkg_query.filter_epoch({pkg.get_epoch()});
```
This should resolve a corner case problem when a package is available in multiple epochs.
Otherwise it looks good!